app/main/authorize/keycloak_login_required_decorator.py

```diff
@@ -80,3 +73,46 @@ def _check_if_user_has_access_to_ayr(keycloak_ayr_user_group, decoded_token):
         if keycloak_ayr_user_group in group:
             group_exists = True
     return group_exists
+
+
+def _decode_token(access_token):
+    keycloak_openid = keycloak.KeycloakOpenID(
+        server_url=current_app.config["KEYCLOAK_BASE_URI"],
+        client_id=current_app.config["KEYCLOAK_CLIENT_ID"],
+        realm_name=current_app.config["KEYCLOAK_REALM_NAME"],
+        client_secret_key=current_app.config["KEYCLOAK_CLIENT_SECRET"],
+    )
+
+    decoded_token = keycloak_openid.introspect(access_token)
+
+    return decoded_token
+
+
+def get_user_transferring_body_groups():
+    """
+    function that returns list of transferring body group names based on assignment of user group in keycloak for a user
+
+    User has user group assigned in keycloak, keycloak user groups are linked to transferring bodies
+    Based on active access token of user, user groups are extracted and list of user groups details retrieved
+
+    Args:
+        N/A
+
+    Returns:
+        function: return list of transferring bodies
```

Comment:
```suggestion
        List[str]: return list of transferring bodies
```

Reply:
reset file to original state : keycloak_login_required_decorator.py